### ISSUE TITLE
Not respecting vim theme

### ISSUE THREAD

**Issue by SilentGlasses** [zed-industries/feedback#894]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I have airline theme in my vim setup and editing in the Zed Terminal does not respect that... 

### Environment

Zed: 0.70.0 (stable)
OS: macOS 12.6.3
Memory: 32 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![Screen Shot 9](https://user-images.githubusercontent.com/22822565/215002337-7ec7bb91-777f-406c-9739-cf54f90f3485.png)

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

No errors in the logs

**Comment by SilentGlasses** [comment]
Is this just because theming is not a thing as yet?    This is not a top priority, just wanted to bubble it up.

**Comment by clo4** [comment]
It's not that the theme isn't respected -- looks like the terminal doesn't support truecolor yet, so terminal tools that output expecting 16.7 million colors instead only get 256. Interestingly if you use `tput colors` you'll see it correctly reports 256, but `COLORTERM` is incorrectly set to `truecolor`.

@JosephTLyons I couldn't see any issues mentioning this issue specifically, but #788 seems like it might be related. Probably worth renaming the issue?

**Comment by mikayla-maki** [comment]
We should support 24 bit color, I remember adding it myself. I wonder what I missed 🤔

**Comment by clo4** [comment]
Yeah, I was just looking and realized that actually yes it does support that! Seems like background highlights are a little wonky.
<img width="1681" alt="Screenshot 2023-02-02 at 11 35 36 am" src="https://user-images.githubusercontent.com/52195359/216201790-2d85209d-8b73-4a9d-aa4b-1ca89edfe0fd.png">

**Comment by mikayla-maki** [comment]
That would make sense, the background rendering was some of the first code I wrote at Zed 😅. It's also not helped because of our massive (and currently unchangeable) line heights. This would track as being a similar issue to #788 as well. Appreciate the research!

**Comment by clo4** [comment]
Actually, looks as simple as the background color is getting reset for every line! Probably visually obvious but I only just noticed... 😅
